### PR TITLE
refactor client side and add histogram chart

### PR DESCRIPTION
**Pull request by zindlerb**
:full_moon_with_face:  :sleepy: :full_moon_with_face:

### CODE REVIEW

File: client/js/components/chart.jsx [hunk 1]
```diff
@@ -1,11 +1,87 @@
...
-    AppStore.getVisualizationData(this.props.schemaName,
-      AppStore.generateChart
-    );
+    AppStore.renderSchema(this.props.schemaName);
+  },
+  statics: {
+    renderTimeSeriesChart: function(table, headerIndex) {
+      headerIndex = headerIndex || 2;
+      if (Array.isArray(table)) {
+        var csvHeader = AppStore.formatCSVHeader(table[0][headerIndex]);
```
Comment (dduugg) [inline client/js/components/chart.jsx]: does `table[0]` exist?

File: client/js/components/chart.jsx [hunk 2]
```diff
@@ -1,11 +1,87 @@
...
+              }
+            }
+          }
+        });
+      }
+    },
+    renderHistogramChart: function(csvData) {
+      var transformHistogramData = function(csvData, headerIndex) {
+        var bucketSize = 2;
+        var header = csvData[0][headerIndex];
```
Comment (dduugg) [inline client/js/components/chart.jsx]: does `csvData[0]` exist?

File: client/js/components/chart.jsx [hunk 3]
```diff
@@ -1,11 +1,87 @@
...
+        var sortedDataPoints = _.sortBy(_.rest(csvData), function(row) { return parseInt(row[headerIndex]); });
```
Comment (dduugg) [inline client/js/components/chart.jsx]: use the radix argument with `parseInt` (it's a security issue).

File: client/js/components/chart.jsx [hunk 4]
```diff
@@ -1,11 +1,87 @@
...
+        var firstPoint = sortedDataPoints[0][headerIndex];
```
Comment (dduugg) [inline client/js/components/chart.jsx]: existential check

File: client/js/components/chart.jsx [hunk 5]
```diff
@@ -1,11 +1,87 @@
...
+        var lastPoint = _.last(sortedDataPoints)[headerIndex];
+
+        var buckets = _.range(firstPoint, lastPoint, bucketSize);          
+
+        var pointIndex = 0;
+        return _.reduce(buckets, function(accum, bucket, ind) {
```
Comment (dduugg) [inline client/js/components/chart.jsx]: full word argument names are more readable/understandable

File: client/js/components/chart.jsx [hunk 6]
```diff
@@ -1,11 +1,87 @@
...
+          accum.push(0);
+          while(sortedDataPoints[pointIndex][headerIndex] < bucket) {
+            accum[ind] += 1;
```
Comment (dduugg) [inline client/js/components/chart.jsx]: not a `++` person?

File: client/js/components/chart.jsx [hunk 7]
```diff
@@ -1,11 +1,87 @@
...
+        var sortedDataPoints = _.sortBy(_.rest(csvData), function(row) { return parseInt(row[headerIndex]); });
+        var firstPoint = sortedDataPoints[0][headerIndex];
+        var lastPoint = _.last(sortedDataPoints)[headerIndex];
+
+        var buckets = _.range(firstPoint, lastPoint, bucketSize);          
+
+        var pointIndex = 0;
+        return _.reduce(buckets, function(accum, bucket, ind) {
+          accum.push(0);
+          while(sortedDataPoints[pointIndex][headerIndex] < bucket) {
```
Comment (dduugg) [inline client/js/components/chart.jsx]: existential check

File: client/js/components/chart.jsx [hunk 8]
```diff
@@ -1,11 +1,87 @@
...
+            //Row Format ['Data Name', num, num, ...]
+            columns: [transformHistogramData(csvData, 2)],
+            type: "bar"
+        },
+        bar: {
+            width: {
+                ratio: 1 // this makes bar width 50% of length between ticks
+            }
+            // or
+            //width: 100 // this makes bar width 100px
```
Comment (dduugg) [inline client/js/components/chart.jsx]: why is this here? what do you think of:

   var width = some_check ? 100 : {ratio: 1}

**Comment by dduugg** [comment]
i think this needs a few existential checks to reduce the chance of runtime errors. otherwise, looks good.

File: client/js/components/chart.jsx (hunk 6 above)
Comment (cmperez) [inline client/js/components/chart.jsx]: crockford advises against ++ in the good parts and I took it to heart

File: client/js/components/chart.jsx (hunk 8 above)
Comment (cmperez) [inline client/js/components/chart.jsx]: I suspect it's leftover inline documentation from copy-pasting :smirk:

**Comment by cmperez** [comment]
:+1: after changes per dug's recommendations
